be/src/exec/pipeline/nljoin/nljoin_probe_operator.cpp

```diff
@@ -186,10 +186,53 @@ void NLJoinProbeOperator::_reset_build_chunk_index() {
     _move_build_chunk_index(0);
 }
 
-void NLJoinProbeOperator::_next_build_chunk_index() {
+void NLJoinProbeOperator::_next_build_chunk_index_for_other_join() {
     _move_build_chunk_index(_curr_build_chunk_index + 1);
 }
 
+// probe one chunk from right
+void NLJoinProbeOperator::_next_probe_row_index_for_inner_join() {
+    // next probe row
+    _probe_row_current++;
+    // probe chunk iter over
+    if (_probe_row_current >= _probe_chunk->num_rows()) {
+        // next build chunk
+        _curr_build_chunk_index++;
+        if (_curr_build_chunk_index >= _num_build_chunks()) {
+            // build chunks iter over
+            _curr_build_chunk_index = 0;
+            _probe_row_current = 0;
+            _probe_chunk = nullptr;
+        } else {
+            // next build chunk
+            _probe_row_current = 0;
+            _curr_build_chunk = _cross_join_context->get_build_chunk(_curr_build_chunk_index);
+            _build_row_current = 0;
+        }
+    }
+}
+
+// probe one chunk from left
+void NLJoinProbeOperator::_next_build_row_index_for_inner_join() {
+    // next build row
+    _build_row_current++;
+    // all rows of current build chunk iter over
+    if (_build_row_current >= _curr_build_chunk->num_rows()) {
+        // next build chunk
+        _curr_build_chunk_index++;
+        _build_row_current = 0;
+        // all build chunks iter over
+        if (_curr_build_chunk_index >= _num_build_chunks()) {
+            _probe_chunk = nullptr;
+        } else {
+            // get next build chunk
+            _curr_build_chunk = _cross_join_context->get_build_chunk(_curr_build_chunk_index);
+        }
+    }
+    // reset probe row index
+    _probe_row_current = 0;
```

Comment:
why reset?

Reply:
It doesn't do anything here, it's just a defense